go/vt/wrangler/reparent.go

```diff
@@ -363,7 +363,11 @@ func (wr *Wrangler) plannedReparentShardLocked(ctx context.Context, ev *events.R
 	if topoproto.TabletAliasEqual(shardInfo.MasterAlias, masterElectTabletAlias) {
 		return fmt.Errorf("master-elect tablet %v is already the master", topoproto.TabletAliasString(masterElectTabletAlias))
 	}
-	oldMasterTabletInfo, ok := tabletMap[*shardInfo.MasterAlias]
+	var oldMasterTabletInfo *topo.TabletInfo
+	ok = false
```

Comment:
I was talking about that, yes. I forgot the special handling.

But in any case, there is just way too much thinking going into reading this code, that depend on non-obvious handling of pointers. How about:

if topoproto.TabletAliasIsZero(shardInfo.MasterAlias) {
    return fmt.Errorf("shard %v/%v has no current master" ...
)

then you don't need the rest of the change. Will be way easier to read and understand.